### PR TITLE
Add collections to publisher metadata if B variant

### PR DESCRIPTION
**Pull request by andrewgarner**
screen shot 2018-08-15 at 11 51 37

https://trello.com/c/kcZ8VQS7

---

Visual regression results:
https://government-frontend-pr-1049.surge.sh/gallery.html

Component guide for this PR:
https://government-frontend-pr-1049.herokuapp.com/component-guide

### CODE REVIEW

**Comment by vanitabarrett** [comment]
This might be a design question: do we want "and 2 others" as well as "+ show all"? I think previously we've combined the two, so "+ 2 others" (for example) becomes the toggle link.

**Comment by andrewgarner** [comment]
When I looked at this with Alan for the top banner we had a couple of thoughts that led to this approach:

1. There is an assumption that the "+" or "-" suggests that the link will show or hide and not take you to another page.
1. "and 2 others" for a hyperlink that hides an element on the page doesn't describe the behaviour.

File: app/views/components/_publisher-metadata.html.erb [hunk 1]
```diff
@@ -16,10 +16,27 @@
...
+              <% if title == :Collections %>
+                <% if values.size <= 2 %>
+                  <%= values.take(2).to_sentence.html_safe %>
+                <% else %>
+                  <% featured_values, *other_values = values %>
+                  <%= featured_values %> and <%= pluralize(other_values.size, 'other') %>
+                  <a href="#"
+                     data-controls="<%= toggle_id %>"
+                     data-expanded="false"
+                     data-toggled-text="&minus; hide all">+ show all</a>
```
[the remark this answers is not in the document]
Comment (andrewgarner) [reply app/views/components/_publisher-metadata.html.erb]: @vanitabarrett I've added translation support with https://github.com/alphagov/government-frontend/pull/1049/commits/48971268e7ade7c4b98a9537cd691d10bb499b30 but we don't have actual translations.

**Comment by vanitabarrett** [comment]
@andrewgarner As this is a change to the publisher metadata component, we probably need to add a test in `publisher_metadata_spec.rb`and an example in the component yml docs (as adding a collection has specific behaviour, especially when there are more than 2)

**Pull request merged**